File: problems/018-worker-generations/solutions/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	in := bufio.NewReader(os.Stdin)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()
	var n int
	var b, used uint64
	fmt.Fscan(in, &n, &b)
	cur := ""
	gen, reject := 0, 0
	for ; n > 0; n-- {
		var f string
		var s uint64
		fmt.Fscan(in, &f, &s)
		if s == 0 {
			fmt.Fprintln(out, "CACHE")
			continue
		}
		if s > 8 || s > b {
			fmt.Fprintln(out, "REJECT")
			reject++
			continue
		}
		if cur != f || used+s > b {
			cur = f
			used = 0
			gen++
		}
		used += s
		fmt.Fprintln(out, "WORKER", gen)
	}
	fmt.Fprintln(out, "SUMMARY", gen, reject)
}

File: problems/018-worker-generations/solutions/javascript/main.js
import * as std from "std";
const a = std.in.readAsString().trim().split(/\s+/);
let p = 0,
  n = Number(a[p++]),
  b = BigInt(a[p++]),
  used = 0n,
  gen = 0,
  reject = 0,
  cur = null,
  out = [];
while (n--) {
  const f = a[p++], s = BigInt(a[p++]);
  if (s === 0n) {
    out.push("CACHE");
    continue;
  }
  if (s > 8n || s > b) {
    out.push("REJECT");
    reject++;
    continue;
  }
  if (cur !== f || used + s > b) {
    cur = f;
    used = 0n;
    gen++;
  }
  used += s;
  out.push(`WORKER ${gen}`);
}
out.push(`SUMMARY ${gen} ${reject}`);
std.out.puts(out.join("\n") + "\n");
